docs/integrations/spark/built_in_lineage.md

```diff
@@ -0,0 +1,224 @@
+  multiple extensions.
+
+* Spark extensions know a lot of valuable metadata that can be published within OpenLineage events.
+  It makes sense to allow extensions publish facets on their own. This [issue](https://github.com/OpenLineage/OpenLineage/issues/167)
+  contains great example of useful aspects that can be retrieved from Iceberg extension.
+
+## Solution
+
+A remedy to the problems above is to migrate lineage extraction logic directly to
+Spark `LogicalPlan` nodes. The advantages of this approach are:
+* **One-to-one version matching** - there is no further need for a single integration code to support
+  multiple versions of a Spark extension.
+* **Avoid breaking changes** - this approach limits amount of upgrades that break integration between
+  `openlineage-spark` and other extensions, as lineage extraction code is directly put into extensions
+  codebase which assures that changes on the Spark extension side are not breaking it.
+* **Ability to publish extension related lineage metadata** - tighter integration level allows Spark
+  extensions to publish more extension-specific lineage metadata.
+
+`spark-interfaces-scala` package contains traits that shall be implemented as well as extra utility
+classes to let integrate OpenLineage within any Spark extension.
+
+Package code should not be shipped with extension that implements traits. Dependency should be marked
+as compile-only. Implementation of the code calling the methods should be responsible for providing
+`spark-interfaces-scala` on the classpath.
+
+Please note that this package as well as the traits should be considered experimental and may evolve
+in the future. All the current logic has been put into `*.scala.v1` package. First, it is possible
+we develop the same interfaces in Java. Secondly, in case of non-compatible changes,
+we are going to release `v2` interfaces. We're aiming to support different versions within spark
+integration.
+
+## Extracting datasets from LogicalPlan nodes
+
+Two interfaces have been prepared:
+* `io.openlineage.spark.builtin.scala.v1.InputLineageNode` with `getInputs` method,
+* `io.openlineage.spark.builtin.scala.v1.OutputLineageNode` with `getOutputs` method.
+
+They return list of `InputDatasetWithFacets` and `OutputDatasetWithFacets` respectively. Each trait has methods
+to expose dataset facets as well facets that relate to particular dataset only in the context of
+current run, like amount of bytes read from a certain dataset.
+
+There are two possible ways to identify dataset returned by a trait. The straightforward one
+is to identify it by `namespace` as `name` strings, which is a classical dataset identifier within
+OpenLineage convention.
+Oftentimes this approach can be cumbersome.
+There are many `command` nodes whose output dataset is `DatasourceV2Relation`, which is another
+node within the logical plan. In this case, it makes
+sense to delegate dataset extraction to other node, while still having the ability to enrich it
```

Comment:
I think it needs more context and explanation about this delegation process. It does not explain what happens when you use it - control goes back to OpenLineage integration and it uses the delegated LogicalPlan node to determine particular dataset. It also does not explain that you depend on that LogicalPlan node to have proper implementation whether using this mechanism or directly in OpenLineage.